### DISCUSSION TITLE
关于普米插件及apisix_http_requests_total数值的问题

### DISCUSSION THREAD

**Discussion by c591762009** [apache/apisix#7374] [Q&A]
有个问题没明白，希望能够解答一下，apisix_http_requests_total这个数值，通过/apisix/prometheus/metrics接口获取，其返回的这个数值，远远超过实际的调用量，这个计算机制是什么？将所有的综合加在一起吗？

**Comment by tzssangglass** [comment]
`apisix_http_requests_total` means: the total number of client requests since APISIX started
ref: https://github.com/apache/apisix/issues/6301 and https://github.com/apache/apisix/pull/6327/files

**Reply by c591762009** [reply]
不好意思，我觉得这个计数是否存在问题。如果按照是所有请求总数来说的话。那么apisix_http_latency_bucket这个指标,依照官方文档所说的“Latency: 每个服务的请求用时和 APISIX 处理耗时的直方图”。那么此时，将我所有的apisix_http_latency_bucket指标的次数进行计算，因为依照官方文档所说的，该是记录每个请求的相关用时，那么包含type为apisix，request，upstream在内。其总和为165.但此时apisix_http_requests_total 的数值变成了1057。并且自启动以来，也没有请求国这么多次。该数值是否存在重复计算的问题？
![image](https://user-images.githubusercontent.com/32794876/177092150-d9af7ab8-5afb-4280-baec-5878c04a3141.png)


![image](https://user-images.githubusercontent.com/32794876/177091437-12d88c21-9136-45cc-8e17-5a4a528b791f.png)
![image](https://user-images.githubusercontent.com/32794876/177091504-055ef673-2944-4cc1-a5ba-3c30c91dc38e.png)
![image](https://user-images.githubusercontent.com/32794876/177091679-2440f07f-36fa-4bbf-a1fa-c3f7bf95a73d.png)
![image](https://user-images.githubusercontent.com/32794876/177091740-0f4511e5-c8b7-4d20-aa6e-b7cd2bddb1c2.png)

**Reply by tzssangglass** [reply]
`apisix_http_latency_bucket` is histograms and not counters. 
example: When the latency of a request is 4, it will appear in these buckets le="5", le="10", le="20", etc. The sum of them is not directly related to the total number of requests.

BTW, when you request `/apisix/prometheus/metrics`, it would also +1 request in `apisix_http_requests_total`.

**Reply by c591762009** [reply]
那在明细中没有体系，我如何清楚，我调用了多少。对于route的调用总和，应该通过什么进行求和

**Reply by tzssangglass** [reply]
we can use `apisix_http_status`
